packages/edit-widgets/src/store/actions.js

```diff
@@ -113,6 +118,17 @@ export function* saveWidgetArea( widgetAreaId ) {
 		return true;
 	} );
 
+	// Get all widgets that have been deleted
+	const deletedWidgets = areaWidgets.filter( ( widget ) => {
+		const { id } = widget;
+		return (
+			widgetsBlocks.findIndex(
+				( { attributes: { __internalWidgetId } } ) =>
+					__internalWidgetId === id
+			) === -1
```

Comment:
Nit: It's clearer (or maybe just more common?) I think to use `! widgetsBlocks.some(...)` instead of `widgetsBlocks.findIndex(...) === -1`.